dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/microprofile/metrics50/http/MPMetricsHTTPMetricsAdapterImpl.java

```diff
@@ -30,10 +40,20 @@
  *
  */
 @Component(service = { HTTPMetricAdapter.class }, configurationPolicy = ConfigurationPolicy.IGNORE)
-public class MPMetricsHTTPMetricsAdapterImpl implements HTTPMetricAdapter {
+public class MPMetricsHTTPMetricsAdapterImpl implements HTTPMetricAdapter, ApplicationStateListener {
 
     static SharedMetricRegistries sharedMetricRegistries;
 
+    private static final TraceComponent tc = Tr.register(MPMetricsHTTPMetricsAdapterImpl.class);
+
+    private static final String NO_APP_NAME_IDENTIFIER = "io.openliberty.microprofile.metrics50.internal.http.no.app.name";
+
+    /**
+     * Mapping between application name to a map of HTTP stats ID mapped to
+     * MicroProfile Metrics' Tags i.e. Map<appName, Map<HttpStatID, Tags>>
+     */
+    private static Map<String, Map<String, Tag[]>> appNameToTagsMap = new ConcurrentHashMap<String, Map<String, Tag[]>>();
```

Comment:
I am a bit concerned about memory leak here causing a DDOS attack.  We had this in the past when trying to cache so many things for different http routes.  You may want to limit the size of this map.

Reply:
#23267 was the PR that had to be reverted where `initDataCache` was created and caused a memory leak with so many different http URIs associated with an application especially with different params in a URI.  If that is not an issue with the cache here, it would not be a problem.

Reply:
The HTTP Routes resolves the direct pattern (i.e. with the asterisk for wildcards in a servlet path (i.e., `/path/*` or the path template itself for a RESTful resource `/path/{param}`. So we should avoid that scenario.

Reply:
@jhanders34 Is it okay to resolve this? (Or I think it needs to be re-reviewed to allow for merging)

Reply:
I approved it.  Did you do tests with servlet and rest to verify that we do what you are saying?  I am concerned about sub resources with rest as well, but maybe we don't do anything with those.  Not sure how you did the rest thing since as far as I know you couldn't get that information.  So you can merge it in now, but I will probably do some extra testing to make sure it is working as expected.

Reply:
Yup there are tests from the original feature delivery. For REST we're using a REST filter (the REST filter that is used is part of the bundle used for registering REST MBeans) to capture the http route information. It's same logic used in the core telemetry code in their REST filter. Similarly, sub resources are not computed (ignored).